src/commands.ts

```diff
@@ -343,6 +362,48 @@ export class CosmosDBCommands {
             }
         }
     }
+
+    public static async deleteDocDBDocument(doc: DocDBDocumentNode, explorer: CosmosDBExplorer): Promise<void> {
+        if (doc) {
+            const confirmed = await vscode.window.showWarningMessage("Are you sure you want to delete document '" + doc.label + "'?", "Yes", "No");
+            if (confirmed === "Yes") {
+                const masterKey = await doc.coll.db.getPrimaryMasterKey();
+                const endpoint = await doc.coll.db.getEndpoint();
+                const client = new DocumentClient(endpoint, { masterKey: masterKey });
```

Comment:
I notice I'm writing these 3 lines a LOT of the time. And I only use the endpoint and masterKey for one purpose - to create a client. 
Low-priority work item to refactor every use into one GenerateClient function (that handles whether a document, collection, db or server is passed)?